lib/src/nanotdf/models/ResourceLocator.ts

```diff
@@ -163,42 +168,40 @@ export default class ResourceLocator {
    * Return the contents of the Resource Locator in buffer
    */
   toBuffer(): Uint8Array {
-    const buffer = new Uint8Array(2 + this.body.length + this.identifierType.valueOf());
-    buffer.set([this.protocol], 0);
-    buffer.set([this.lengthOfBody], 1);
-    buffer.set(new TextEncoder().encode(this.body), 2);
+    const buffer = new Uint8Array(
+      ResourceLocator.BODY_OFFSET + this.body.length + this.identifierType
+    );
+    let idTypeNibble = 0;
+    switch (this.identifierType) {
+      case ResourceLocatorIdentifierEnum.TwoBytes:
+        idTypeNibble = ResourceLocator.IDENTIFIER_2_BYTE;
+        break;
+      case ResourceLocatorIdentifierEnum.EightBytes:
+        idTypeNibble = ResourceLocator.IDENTIFIER_8_BYTE;
+        break;
+      case ResourceLocatorIdentifierEnum.ThirtyTwoBytes:
+        idTypeNibble = ResourceLocator.IDENTIFIER_32_BYTE;
+        break;
+    }
+    buffer.set([this.protocol | idTypeNibble], ResourceLocator.PROTOCOL_OFFSET);
+    buffer.set([this.lengthOfBody], ResourceLocator.LENGTH_OFFSET);
+    buffer.set(new TextEncoder().encode(this.body), ResourceLocator.BODY_OFFSET);
     if (this.identifier) {
-      buffer.set(new TextEncoder().encode(this.identifier), 2 + this.body.length);
+      buffer.set(
+        new TextEncoder().encode(this.identifier),
+        ResourceLocator.BODY_OFFSET + this.body.length
+      );
     }
     return buffer;
   }
 
-  /**
-   * Get URL
-   *
-   * Construct URL from ResourceLocator or throw error
-   */
-  getUrl(): string | never {
-    let protocol: string;
-    // protocolIndex get the first four bits
-    const protocolIndex: number = this.protocol & 0xf;
-    if (protocolIndex === ProtocolEnum.Http) {
-      protocol = 'http';
-    } else if (protocolIndex === ProtocolEnum.Https) {
-      protocol = 'https';
-    } else {
-      throw new Error(`Cannot create URL from protocol, "${ProtocolEnum[this.protocol]}"`);
-    }
-    return `${protocol}://${this.body}`;
-  }
-
   /**
    * Get Identifier
    *
    * Returns the identifier of the ResourceLocator or an empty string if no identifier is present.
    * @returns { string } Identifier of the resource locator.
    */
-  getIdentifier(): string {
-    return this.identifier || '';
+  get kid(): string {
```

Comment:
Identifier is recommended  since it is used for lookups of KAS key, Remote Policy, Policy key

see https://github.com/opentdf/spec/blob/60171bc24301bf696462140222f972e697a8ae38/schema/nanotdf/README.md#341-resource-locator